src/Microsoft.Extensions.Primitives/StringValues.cs

```diff
@@ -27,8 +27,23 @@ public StringValues(string value)
 
         public StringValues(string[] values)
         {
-            _value = null;
-            _values = values;
+            if (values == null || values.Length == 0)
```

Comment:
Maybe add a comment explaining that if _values isn't set to null when values.Length is zero, the IsNull property will be broken.

Reply:
Added